include/psa/crypto_struct.h

```diff
@@ -228,19 +277,17 @@ struct psa_key_derivation_s
     size_t capacity;
     union
     {
-        struct
-        {
-            uint8_t *data;
-            size_t size;
-        } buffer;
+        /* Make the union non-empty even with no supported algorithms. */
+        uint8_t dummy;
```

Comment:
Is this necessary? Why cant we have the union guarded by `#if defined(MBEDTLS_MD_C)`? That way we could remove the dummy

Reply:
That is an option too. I decided to go with the dummy member, because this way
- we don't need to have ifdefs in the macro initialiser (`PSA_KEY_DERIVATION_OPERATION_INIT`)
- it is easier to introduce new members in the union that depend on other compile time flags
- we already use this solution with other similar structs in the library

Reply:
It's just a coincidence that all the KDF we have now use a hash internally. We're likely to add something else soon, such as NIST KDF built on CMAC.

Reply:
ok